### PR TITLE
Use a slack app instead of manual slack configuration

### PR DESCRIPTION
**Pull request by Connoropolous**
check out https://github.com/Hylozoic/hylo-node/pull/81 for images

### CODE REVIEW

File: src/html/ui/community/settings.tpl.html [hunk 1]
```diff
@@ -213,5 +213,29 @@
...
+    <div class="section-label" ng-click="expand5 = !expand5">
+      Send Updates to Slack
+      <i ng-class="{'icon-down': expand5, 'icon-right': !expand5}"></i>
+    </div>
+    <div class="section slack" collapse="!expand5">
+      <div class="setting-item">
+        <div class="full-column">
+          <div class="add-to-slack" ng-hide="community.slack_hook || slack_remove_transition">
+            <p>Connect this community to a <a href="https://slack.com" target="_blank">Slack</a> team and Hylo will notify a channel when there are new posts.</p>
+            <a href="https://slack.com/oauth/authorize?scope=incoming-webhook&client_id=15397411765.20785715792&redirect_uri={{add_slack_url}}"><img alt="Add to Slack" height="40" width="139" src="https://platform.slack-edge.com/img/add_to_slack.png" srcset="https://platform.slack-edge.com/img/add_to_slack.png 1x, https://platform.slack-edge.com/img/add_to_slack@2x.png 2x"></a>
```
Comment (Connoropolous) [inline src/html/ui/community/settings.tpl.html]: Aha, client_id should be changed to an environment variable no doubt

File: src/js/app/controllers/community/CommunitySettingsCtrl.js [hunk 2]
```diff
@@ -5,9 +5,16 @@ var controller = function ($scope, $history, $analytics, community, currentUser,
   _.merge(community, extraProperties);
   var origin = $location.absUrl().replace($location.path(), '');
   $scope.join_url = origin + '/c/' + community.slug + '/join/' + community.beta_access_code;
+  $scope.add_slack_url = origin + '/noo/community/' + community.id + '/settings/slack';
   $scope.community = community;
   $scope.settings = community.settings;
 
+  if ($location.$$search.slack === "1") {
```
Comment (levity) [inline src/js/app/controllers/community/CommunitySettingsCtrl.js]: huh, i didn't know about `$$search`; that would have saved me some time in the past :)

**Comment by levity** [comment]
this looks pretty good, modulo your comment about pulling client id into an env var. 

btw, speaking of front-end matters, i want to draw your attention to [hylo-redux](/Hylozoic/hylo-redux), which is a rewrite of the frontend in React and Redux. it started out as an experiment and quickly started snowballing. :) it's live at [next.hylo.com](http://next.hylo.com).

File: src/js/app/controllers/community/CommunitySettingsCtrl.js (hunk 2 above)
Comment (Connoropolous) [inline src/js/app/controllers/community/CommunitySettingsCtrl.js]: I discovered it by logging the $location object to see if there was any easy way to get query string params.

**Comment by Connoropolous** [comment]
Yes! exciting about the hylo-redux thing. I had actually already gotten to cloning it and taking a look. Neat to see the one in action with content though! going to be great

File: src/html/ui/community/settings.tpl.html (hunk 1 above)
Comment (Connoropolous) [inline src/html/ui/community/settings.tpl.html]: Actually @levity I'm a little bit lost as to how to actually get an environment variable into this strign

**Comment by Connoropolous** [comment]
ok, this is set to match the backend column name change now.

**Comment by Connoropolous** [comment]
@levity can you help me out with an idea for how to replace the CLIENT_ID var in the one file? it's all that's left to do

**Comment by levity** [comment]
There are basically two pieces to this. [_clientEnv.ejs](https://github.com/Hylozoic/hylo-frontend/blob/master/src/html/pages/_clientEnv.ejs) is a file that exports selected environment variables into a global JS object on the client. So you'll want to add an item in there. Then you'll want to modify [deploy.js](https://github.com/Hylozoic/hylo-frontend/blob/master/deploy.js#L145) so that it knows to get the variable from the Heroku environment.

**Comment by levity** [comment]
By the way, does it make sense for us to just use the Slack app you've set up (assuming we can change the client ID or that it's not an issue that it's been exposed), or should we create our own?

**Comment by Connoropolous** [comment]
Ok @levity those changes are made now. 

we could just use the app that I've created. I might request some support on the wordsmithing for different description fields. I don't think it's particularly an issue that the client id was temporarily exposed. there's nothing significant that can be faked without the client secret too

**Pull request merged**